src/Datadog.Trace/ExtensionMethods/DictionaryExtensions.cs

```diff
@@ -79,5 +79,7 @@ public static bool TryGetValue<TValue>(this IDictionary dictionary, object key,
 
             return dictionary.Count == 0;
         }
+
+        public static bool IsNullOrEmpty<TKey, TValue>(this IDictionary<TKey, TValue> dictionary) => dictionary?.IsEmpty() ?? true;
```

Comment:
Is it just me that hates the "negative" of `IsNullOrEmpty()`, especially as we always do `!string.IsNullOrEmpty()`?!

I wish `string.HasValue()` was a thing, so that this would be `HasValue()` 😄

Reply:
@andrewlock Just add an extension method with aggressive inlining if that makes you feel better 😄